File: DEP-ui/src/app/models/player-response-dto.ts
export interface PlayerResponseDto {
  id: number; 
  username: string;
  email: string;

  stats: Stats;
  inventory: Inventory;
  lastPosition: LastPosition;
}

export interface LastPosition {
  playerId: number;
  x: number;
  y: number;
  mapId: number;
  angle: number;
}

export interface Stats {
  id: number; 
  level: number;
  experience: number;
  honor: number;
  companyPoints: number;
}

export interface Inventory {
  id: number;
}

File: DEP-REST/src/main/java/com/sparklecow/dark_engine_protocol/models/PlayerResponseDto.java
package com.sparklecow.dark_engine_protocol.models;

import com.sparklecow.dark_engine_protocol.entities.Inventory;
import com.sparklecow.dark_engine_protocol.entities.LastPosition;
import com.sparklecow.dark_engine_protocol.entities.Stats;

public record PlayerResponseDto(
        Long id,
        String username,
        String email,
        Stats stats,
        Inventory inventory,
        LastPosition lastPosition
) {
}